src/DurableSDK/PowerShellServices.cs

```diff
@@ -6,50 +6,199 @@
+            }
+
+            // assign SetFunctionInvocationContextCommand to the corresponding external SDK's CmdLet
+            SetFunctionInvocationContextCommand = string.Format(
+                PowerShellWorkerStrings.SetFunctionInvocationContextCmdLetTemplate,
+                PowerShellWorkerStrings.ExternalDurableSDKName);
+        }
+
+        public bool HasExternalDurableSDK()
+        {
+            return _usesExternalDurableSDK;
+        }
+
+        public PowerShell GetPowerShell()
+        {
+            return this._pwsh;
         }
 
         public void SetDurableClient(object durableClient)
         {
             _pwsh.AddCommand(SetFunctionInvocationContextCommand)
                 .AddParameter("DurableClient", durableClient)
                 .InvokeAndClearCommands();
-
-            _hasSetOrchestrationContext = true;
+            _hasInitializedDurableFunctions = true;
         }
 
-        public void SetOrchestrationContext(OrchestrationContext orchestrationContext)
+        public OrchestrationBindingInfo SetOrchestrationContext(
+            ParameterBinding context,
+            out IExternalOrchestrationInvoker externalInvoker)
         {
-            _pwsh.AddCommand(SetFunctionInvocationContextCommand)
-                .AddParameter("OrchestrationContext", orchestrationContext)
-                .InvokeAndClearCommands();
+            externalInvoker = null;
+            OrchestrationBindingInfo orchestrationBindingInfo = new OrchestrationBindingInfo(
+                context.Name,
+                JsonConvert.DeserializeObject<OrchestrationContext>(context.Data.String));
+
+            if (_usesExternalDurableSDK)
+            {
+                Collection<Func<PowerShell, object>> output = _pwsh.AddCommand(SetFunctionInvocationContextCommand)
+                    // The external SetFunctionInvocationContextCommand expects a .json string to deserialize
+                    // and writes an invoker function to the output pipeline.
+                    .AddParameter("OrchestrationContext", context.Data.String)
+                    .InvokeAndClearCommands<Func<PowerShell, object>>();
 
-            _hasSetOrchestrationContext = true;
+                // If more than 1 element is present in the output pipeline, we cannot trust that we have
+                // obtained the external orchestrator invoker; i.e the output contract is not met.
+                var numResults = output.Count();
+                var numExpectedResults = 1;
+                var outputContractIsMet = output.Count() == numExpectedResults;
+                if (outputContractIsMet)
+                {
+                    externalInvoker = new ExternalInvoker(output[0]);
+                }
+                else
+                {
+                    var exceptionMessage = string.Format(PowerShellWorkerStrings.UnexpectedResultCount,
```

Comment:
Should we log the module names and versions?

Reply:
Yes! But we actually already do that earlier on, when we succeed in importing the DF SDK. With that in mind, do we really have to log it again?

Reply:
For context, we log that here: https://github.com/Azure/azure-functions-powershell-worker/blob/7c24933ef941879b617fd69be90814cc5c3e45fe/src/DurableSDK/PowerShellServices.cs#L59-L62